### PR TITLE
Add automated data evaluation - metrics and plots

### PR DESCRIPTION
**Pull request by kuba1302**
More things to do: 
Per class precision and recall 
Partial feature importance plot

### CODE REVIEW

File: evaluator.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        if save_path:
+            fig.savefig(save_path / "precision_recall_curve.png")
+
+        return fig
+
+    def plot_partial_dependency_plot(
+        self, save_path: Path = None, kind: str = "average"
+    ) -> Figure:
+
+        cols = [col for col in self.X.columns]
```
Comment (jzmujdzin) [inline evaluator.py]: why not just use self.X.columns?

File: evaluator.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        pdp = PartialDependenceDisplay.from_estimator(
+            self.model, self.X, cols, kind=kind
+        )
+
+        number_of_cols = len(cols)
+        fig, ax = plt.subplots(1, 1)
+        fig.set_size_inches(1 * number_of_cols, 1.5 * number_of_cols)
```
Comment (jzmujdzin) [inline evaluator.py]: why not use len(cols)? i believe its self explanatory

File: evaluator.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        if save_path:
+            fig.savefig(save_path / "roc_curve.png")
+
+        return fig
+
+    def plot_precision_recall_curve(self, save_path: Path = None) -> Figure:
+        precision, recall, thresholds = precision_recall_curve(
+            self.y, self.proba
+        )
+        auc_score = auc(recall, precision)
```
Comment (jzmujdzin) [inline evaluator.py]: self.get_pr_auc() calculates the same thing

File: evaluator.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        if save_path:
+            fig.savefig(save_path / "precision_recall_curve.png")
+
+        return fig
+
+    def plot_partial_dependency_plot(
+        self, save_path: Path = None, kind: str = "average"
+    ) -> Figure:
+
+        cols = self.X.columns
```
Comment (jzmujdzin) [inline evaluator.py]: should we assign another variable for this particular use?

File: evaluator.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+        pdp = PartialDependenceDisplay.from_estimator(
+            self.model, self.X, cols, kind=kind
+        )
+
+        fig, ax = plt.subplots(1, 1)
+        fig.set_size_inches(1 * len(cols), 1.5 * len(cols))
```
Comment (jzmujdzin) [inline evaluator.py]: same here, wouldn't len(self.X.columns) be sufficient? what do you think?

**Review by jzmujdzin — APPROVED** [review]

File: evaluator.py (hunk 4 above)
Comment (kuba1302) [reply evaluator.py]: hm i think it's not that readable to uses that long expression many times

File: evaluator.py (hunk 5 above)
Comment (kuba1302) [reply evaluator.py]: same as above

**Pull request merged**